### PR TITLE
GA4: set custom dimensions

### PR DESCRIPTION
**Pull request by aburke07**
## Overview
Adds the preferred store and preferred language (always english in react-commerce).

## QA
1. Apply this diff _before running `pnpm install:all`_
<details><summary>Diff</summary>
<p>

```diff
NEXT_PUBLIC_GA_KEY=UA-30506-1
 # If we want to test Google Analytics in dev DEBUG should be true, GTAG_ID should be G-5ZXNWJ73GK, and URL should be
 # https://www.google-analytics.com/analytics_debug.js
-NEXT_PUBLIC_GA_DEBUG=
-NEXT_PUBLIC_GA_URL=
-NEXT_PUBLIC_GTAG_ID=
+NEXT_PUBLIC_GA_DEBUG=true
+NEXT_PUBLIC_GA_URL=https://www.google-analytics.com/analytics_debug.js
+NEXT_PUBLIC_GTAG_ID=G-5ZXNWJ73GK
 NEXT_PUBLIC_ALGOLIA_PRODUCT_INDEX_NAME=dev_product_group_en
 NEXT_PUBLIC_DEFAULT_STORE_CODE=test
```

</p>
</details> 

2. Make sure the custom dimensions show up in [GA4 debugger](https://analytics.google.com/analytics/web/#/a30506p383872824/admin/debugview/overview) when an event is fired.
3. The language dimension should now match the language of your logged in user on Cominor.

Closes https://github.com/iFixit/react-commerce/issues/2019
Closes https://github.com/iFixit/react-commerce/issues/2018
Connects https://github.com/iFixit/ifixit/issues/50135

### CODE REVIEW

**Comment by deltuh-vee** [comment]
QA 🎬 
CleanShot 2023-10-17 at 10 06 36
Custom dimensions are now set in GA4.

**Comment by aburke07** [comment]
Dev_block :alien: on possibly not hardcoding the language

File: frontend/components/analytics/GoogleAnalytics.tsx [hunk 1]
```diff
@@ -39,6 +40,7 @@ function GA4() {
    const { query } = router;
 
    const debugMode = GA_DEBUG || query.ga4_debug === 'true';
+   setupMinimumGA4(GTAG_ID, debugMode);
```
Comment (danielcliu-ifixit) [inline frontend/components/analytics/GoogleAnalytics.tsx:43]: did you run into issues when it wasn't used?

**Comment by danielcliu-ifixit** [comment]
CR :v:

File: frontend/components/analytics/GoogleAnalytics.tsx (hunk 1 above)
Comment (aburke07) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: Yeah the events just wouldn't send to GA4 🤷

Comment (danielcliu-ifixit) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: hmm i hadn't run into that. the code in 
```
window.dataLayer = window.dataLayer || [];
               function gtag(){dataLayer.push(arguments);}
               gtag('js', new Date());
               
```
               
should have set it up in time

Comment (aburke07) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: Yep. I played around with that code for a while and couldn't get anything to send. Calling `setupMinimumGA4()` finally made it work with no drawbacks, so I stuck with it.

Comment (danielcliu-ifixit) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: hmmm, im slightly worried about redundancy/spaghetti but we seem to be transitioning to App routing pages which is what `setupMinimumGA4` was made to handle anyway, so its probably to sweep this under the rug

Comment (sctice-ifixit) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: @jarstelfox Do you have any insight here? I think we should understand why this works before we run with it.

Comment (jarstelfox) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: No insights, but I'd try not to call `setupMinimumGA4` it effectively is repeating the code over again

Comment (danielcliu-ifixit) [reply frontend/components/analytics/GoogleAnalytics.tsx:43]: we could delete the script in `GoogleAnalytics` to reduce the redundancy

>I think we should understand why this works before we run with it.

I would say the question is more precisely "Why didn't the code that was working before now not work without calling `setupMinimumGA4()`?".

Comment (aburke07) [reply frontend/components/analytics/GoogleAnalytics.tsx]: Refactored and removed the duplicate js from the `<head>` in [a116c6e](https://github.com/iFixit/react-commerce/pull/2029/commits/a116c6ebcf6e2741ab623a36b4b888c193d8935e).

**Comment by aburke07** [comment]
Un_dev_block :alien: Fixed CR comment and we no longer hardcode the language. 

QA team: the language dimension should now match the language of your logged in user on Cominor.

File: frontend/components/analytics/GoogleAnalytics.tsx [hunk 2]
```diff
@@ -38,30 +38,19 @@ export function GoogleAnalytics() {
 function GA4() {
    const router = useRouter();
    const { query } = router;
+   const dimensions = useGACustomDimensions();
+
+   console.log('dimensions', dimensions);
```
Comment (danielcliu-ifixit) [inline frontend/components/analytics/GoogleAnalytics.tsx:42]: console log

File: packages/analytics/google/index.tsx [hunk 3]
```diff
@@ -79,14 +79,15 @@ function windowGtag() {
...
    GtagID: string | undefined,
-   debugMode: boolean
+   debugMode: boolean,
+   dimensions: GACustomDimensions
 ) {
    if (typeof window !== 'undefined' && GtagID) {
       window.dataLayer = window.dataLayer || [];
       window.gtag = windowGtag;
       window.gtag('js', new Date());
       window.gtag('config', GtagID, debugMode ? { debug_mode: true } : {});
```
Comment (danielcliu-ifixit) [inline packages/analytics/google/index.tsx:89]: i would be careful running this mulitple times per page load. In my experience you can alter the config after the first time for a given page load. Combined with the linter 
`React Hook React.useEffect has a missing dependency: 'debugMode'. Either include it or remove the dependency array`
We may want to just update the dimensions with a useEffect?

**Comment by danielcliu-ifixit** [comment]
dev_block :v:  on concerns about the useEffect usage

File: frontend/components/analytics/GoogleAnalytics.tsx (hunk 2 above)
Comment (aburke07) [reply frontend/components/analytics/GoogleAnalytics.tsx]: [dfbd9c6](https://github.com/iFixit/react-commerce/pull/2029/commits/dfbd9c6cd03176809d595ebc728bba8865f145a9)

**Comment by aburke07** [comment]
Un_dev_block :alien:

**Comment by deltuh-vee** [comment]
QA 🎬 
`preferred_language` now matches the language for the logged in user.

**Comment by danielcliu-ifixit** [comment]
CR :v:

**Review by danielcliu-ifixit — APPROVED** [review]

**Pull request merged**